prov/net/src/xnet_progress.c

```diff
@@ -811,6 +706,53 @@ static ssize_t xnet_recv_hdr(struct xnet_ep *ep)
 	return ep->cur_rx.handler(ep);
 }
 
+static void xnet_complete_rx(struct xnet_ep *ep, ssize_t ret)
+{
+	struct xnet_xfer_entry *rx_entry;
+	struct util_cq *cq;
+
+	rx_entry = ep->cur_rx.entry;
+	if (!rx_entry) {
```

Comment:
When would this if statement be true?

Reply:
Wouldn't that case result in xnet_recv_hdr() returning -FI_EAGAIN?  See comment below.